# HG changeset patch
# User Rik <rik@octave.org>
# Date 1427826692 25200
#      Tue Mar 31 11:31:32 2015 -0700
# Node ID 0376c12256607013b3847ce3d9b4deeebb9ac2d9
# Parent  91e2212b060d3c64d9cd8bb26f7820729517b461
Fix ginput to return button number with Qt toolkit (bug #44676).

* Canvas.cc (canvasMousePressEvent): Supply "button_number (event)" as first
argument when running post_callback() for "windowbuttondownfcn".

diff --git a/libgui/graphics/Canvas.cc b/libgui/graphics/Canvas.cc
--- a/libgui/graphics/Canvas.cc
+++ b/libgui/graphics/Canvas.cc
@@ -588,20 +588,22 @@ Canvas::canvasMousePressEvent (QMouseEve
         {
         case NoMode:
           gh_manager::post_set (figObj.get_handle (), "selectiontype",
                                 Utils::figureSelectionType (event), false);
 
           updateCurrentPoint (figObj, obj, event);
 
           gh_manager::post_callback (figObj.get_handle (),
-                                     "windowbuttondownfcn");
+                                     "windowbuttondownfcn",
+                                     button_number (event));
 
           gh_manager::post_callback (currentObj.get_handle (),
-                                     "buttondownfcn", button_number (event));
+                                     "buttondownfcn",
+                                     button_number (event));
 
           if (event->button () == Qt::RightButton)
             ContextMenu::executeAt (currentObj.get_properties (),
                                     event->globalPos ());
           break;
 
         case TextMode:
           // Handle text insertion here.
